src/subscription_manager/repolib.py

```diff
@@ -437,6 +437,7 @@ def perform(self):
                     if existing is None:
                         repo_file.add(fixed_cont)
                         self.report_add(fixed_cont)
+                        server_value_repo_file.update(fixed_cont)
```

Comment:
I'm not sure what this is for, and how it relates to the PR. Can you please explain this change a bit more?

Reply:
During a "re-register --force" the "existing is None" is true but "server_value_repo is None" is false. Therefore, the repo_file will be set accordingly. Additionally, the server_value_repo need to be updated. 

I did not find a description of the server_value_repo. Maybe I misunderstood it. I guess, it need to be "in sync" with the latest changes from server. 

Currently, the debian sub-man still has an issue that after a refresh the certs are sometimes wrong. Means, the certs given in "URIs" is different to the ssclientcert/sslclientkey certs:

> sslclientcert: /etc/pki/entitlement/2105303304298314886.pem
> URIs: katello://5724319104074315656@test-server[...]

So, somewhere in the refresh logic, the fix_content is not used OR the old value (or not updated) value of server_value_repo is used.

Reply:
Thanks for the explanation; I'm still new to sub-man, so I don't know that part well enough to comment. In any case, I recommend to move this change as separate PR, because it is a different and mostly unrelated change from the `AptRepoFile` fix (and thus will need a separate discussion).